### DISCUSSION TITLE
Dynamic Pages that use getStaticPaths failing at build time

### DISCUSSION THREAD

**Discussion by danboyle8637** [vercel/next.js#13687] [Help]
I am using both getStaticPaths and getStaticProps to query and pull in data from my Sanity backend.

In development, everything works great. 

But when I build my site, my props are all undefined for my dynamic pages only... so clearly the query is not being run.

I think this has to do with my environment variables not being read correctly. In development, I have a .env.development file and I have a similar .env.production to build.

In Vercel, I have my git repo hooked up and I have entered my keys and secrets and endpoints in the environment variables section under the production tab.

I am not using a next.config.js file.

I've looked through the Sanity example and while I do a few things different in regards to getting the data... my getStaticPaths and getStaticProps methods are essentially the same.

All of my errors are a TypeError saying I can't destructure "variable" from undefined.

Anybody run into this... if not I will create a sharable example because I feel like I've tried everything I can think of.

Thanks!

**Comment by Vadorequest** [comment]
@danboyle8637 Hard to help you troubleshoot without code examples/repro.

My first guess is that you should log your variables, especially those you're trying to destructure. (and you could use something like `const { a, b } = myVar || {}` to avoid destructuring failure at runtime.

Take a look at the demo/code below, it'll show you how we do it using NRN:

https://github.com/UnlyEd/next-right-now

Demo: https://nrn-v2-mst-aptd-gcms-lcz-sty-c1.now.sh

Source code: https://github.com/UnlyEd/next-right-now/blob/v2-mst-aptd-gcms-lcz-sty/src/pages/%5Blocale%5D/examples/native-features/example-with-ssg.tsx

It also uses the recent `.env` files.

**Reply by danboyle8637** [reply]
@Vadorequest Thanks I did give an empty object if my variable was undefined. And I got my site to build.

However, know I can't get the build process to use my Sanity configuration.

But thank you for the help because it seems that might be fixed although I can't really tell until I can see my data in my site.
